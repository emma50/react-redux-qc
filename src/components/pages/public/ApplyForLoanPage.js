import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { useForm } from "react-hook-form";
import { joiResolver } from '@hookform/resolvers';
import { ToastContainer} from 'react-toastify';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import validateLoan from '../../../helpers/validation/loans'; 
import { userApplyForLoan } from '../../../actions/loans';

function Copyright() {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {'Copyright Â© '}
      <Link color="inherit" href="https://material-ui.com/">
        Your Website
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  error: {
    background: 'yellow',
  },
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

export default function ApplyForLoan() {
  const classes = useStyles();
  const { register, handleSubmit, errors, } = useForm({
    resolver: joiResolver(validateLoan),
  });
  const dispatch = useDispatch();
  
  const onSubmit = (data, e) => {
    e.preventDefault();
    dispatch(userApplyForLoan(data));
  }

  return (
    <div>
      <ToastContainer/>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <Avatar className={classes.avatar}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Apply for loan
          </Typography>
          <form className={classes.form} onSubmit={handleSubmit(onSubmit)} noValidate>
            {errors.amount  && <Typography className={classes.error}>{errors.amount.message}</Typography>}
            {errors.tenor && <Typography className={classes.error}>{errors.tenor.message}</Typography>}
            <TextField
              inputRef={register}
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="amount"
              label="Amount"
              placeholder="e.g 10000"
              name="amount"
              autoComplete="amount"
              autoFocus
            />
            <TextField
              inputRef={register}
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="tenor"
              label="Tenor"
              placeholder="e.g 1-12"
              type="tenor"
              id="tenor"
              autoComplete="tenor"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Apply
            </Button>
            <Grid container>
              <Grid item>
                <Link href="/signup" variant="body2">
                  {"Don't have an account? Sign Up"}
                </Link>
              </Grid>
            </Grid>
          </form>
        </div>
        <Box mt={8}>
          <Copyright />
        </Box>
      </Container>
    </div>
  );
}

ApplyForLoan.propTypes = {
  validateLoan: PropTypes.object,
  userApplyForLoan: PropTypes.func,
  amount: PropTypes.string,
  tenor: PropTypes.string
}
