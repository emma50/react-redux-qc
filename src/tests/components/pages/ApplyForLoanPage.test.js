import React from 'react';
// import { render, screen } from '@testing-library/react'; 
import renderer from 'react-test-renderer';
import ApplyForLoanPage from '../../../components/pages/public/ApplyForLoanPage';


test('create ApplyForLoanPage snapshot', () => {
  const component = renderer.create(
    <ApplyForLoanPage/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
