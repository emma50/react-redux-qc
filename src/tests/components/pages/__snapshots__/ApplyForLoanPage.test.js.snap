// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create ApplyForLoanPage snapshot 1`] = `
<div>
  <div
    className="Toastify"
  />
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      className="makeStyles-paper-2"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-3 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
          />
        </svg>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Apply for loan
      </h1>
      <form
        className="makeStyles-form-4"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="amount"
            id="amount-label"
          >
            Amount
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="amount"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="amount"
              name="amount"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="e.g 10000"
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-8"
              >
                <span>
                  Amount
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="tenor"
            id="tenor-label"
          >
            Tenor
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="tenor"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="tenor"
              name="tenor"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="e.g 1-12"
              required={true}
              type="tenor"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-8"
              >
                <span>
                  Tenor
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-5 MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Apply
          </span>
        </button>
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="/signup"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Don't have an account? Sign Up
            </a>
          </div>
        </div>
      </form>
    </div>
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
      >
        Copyright © 
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="https://material-ui.com/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Your Website
        </a>
         
        2020
        .
      </p>
    </div>
  </main>
</div>
`;
